rewrite-java-core/src/test/java/com/azure/recipes/core/v2/TypeReferenceTest.java

```diff
@@ -0,0 +1,148 @@
+package com.azure.recipes.core.v2;
+
+import org.intellij.lang.annotations.Language;
+import org.junit.jupiter.api.Test;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+/**
+ * TypeReferenceTest is used to test out the recipe that changes the usage of TypeReference (azure core v1)
+ * to ParameterizedType (azure core v2)
+ * @author Ali Soltanian Fard Jahromi
+ */
+public class TypeReferenceTest implements RewriteTest {
+
+    /**
+     * This method sets which recipe should be used for testing
+     * @param spec stores settings for testing environment; e.g. which recipes to use for testing
+     */
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipeFromResource("/META-INF/rewrite/rewrite.yml",
+                "com.azure.rewrite.java.core.MigrateAzureCoreSamplesToAzureCoreV2");
+    }
+
+    /**
+     * This test method is used to make sure that TypeReference is correctly
+     * changed to ParameterizedType when using List generic type
+     */
+    @Test
+    void testTypeReferenceVariableDeclarationChangeList() {
+        @Language("java") String before = "";
+        before += "\nimport java.lang.reflect.ParameterizedType;";
+        before += "\nimport java.lang.reflect.Type;";
+        before += "\nimport com.azure.core.util.serializer.TypeReference;";
+        before += "\npublic class Testing {";
+        before += "\n  private static final TypeReference<java.util.List<String>> TESTING_TYPE = new TypeReference<java.util.List<String>>() {\n  };";
+        before += "\n}";
+
+
+        @Language("java") String after = "import java.lang.reflect.ParameterizedType;\n" +
+                "import java.lang.reflect.Type;\n" +
+                "public class Testing {\n" +
+                "  private static final java.lang.reflect.Type TESTING_TYPE = new ParameterizedType() {\n" +
+                "      @Override\n" +
+                "      public java.lang.reflect.Type getRawType() {\n" +
+                "          return java.util.List.class;\n" +
```

Comment:
We are still using fully qualified names here and other places.

Earlier comment https://github.com/Azure/azsdk-java-rewrite-recipes/pull/14#discussion_r1750823950

Reply:
Addressed in latest commit. Fully qualified names no longer used when imports are already added.